### PR TITLE
[RFC][UN-14898] Update to ember-cli 3.5

### PR DESCRIPTION
**Pull request by migueldemoura**
### JIRA ticket:
https://uniplaces.atlassian.net/browse/UN-14898

### Context
Lots of changes, please check them carefully.

- Updates ember-cli from 2.18 to 3.5
- Modernizes the whole test suite
- Removes a bunch of deprecations
- Adds hbs template linting and fixes most of the errors
- Partially removes jQuery dependency

### CODE REVIEW

File: addon/styles/partials/_uni-datepicker-range.scss [hunk 1]
```diff
@@ -73,13 +73,13 @@ $desktop-datepicker-width: 260px;
 
             &--range-start,
             &--first-highlighted {
-                background-image: url(assetsRootURL() + "assets/images/ember-cli-uniq/first-highlighted.svg");
```
Comment (AndreJoaquim) [inline addon/styles/partials/_uni-datepicker-range.scss]: Revert this change. We need this for all the projects

**Review by mmelo — APPROVED** [review]

**Review by tlfbrito — APPROVED** [review]

**Review by AndreJoaquim — COMMENTED** [review]
@migueldemoura  and @tiagobrito: This cannot be merged, at least not soon.

This PR provides breaking changes that make this add-on not compatible with ember applications with ember's version less than 3.x, i.e., most of our front-end platforms

**Comment by migueldemoura** [comment]
@AndreJoaquim, fixed with the latest commit.

**Comment by tlfbrito** [comment]
@AndreJoaquim can you check this again ?

**Comment by migueldemoura** [comment]
No regressions on the other older projects. This PR should probably be included in the same version, though: https://github.com/uniplaces/ember-cli-uniq/pull/197. We might as well address these issues too: https://github.com/uniplaces/ember-cli-uniq/issues/265, https://github.com/uniplaces/ember-cli-uniq/issues/208.

This is BC nevertheless as it requires ember-responsive 3.x. Tests using the responsive helper of this version must be in the new qunit format.
